dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -749,20 +753,90 @@ def test_json_encode_after_update(self):
             {
-                "name": None,
-                "col_index": np.nan,
-                "sample_size": 12,
-                "metadata": {},
-                "times": {"categories": 0.0},
-                "thread_safe": True,
-                "_categories": {"c": 5, "b": 4, "a": 3},
-                "_CategoricalColumn__calculations": {},
-                "_top_k_categories": None,
-            },
+                "class": "CategoricalColumn",
+                "data": {
+                    "name": None,
+                    "col_index": np.nan,
+                    "sample_size": 12,
+                    "metadata": {},
+                    "times": {"categories": 0.0},
+                    "thread_safe": True,
+                    "_categories": {"c": 5, "b": 4, "a": 3},
+                    "_CategoricalColumn__calculations": {},
+                    "_top_k_categories": None,
+                },
+            }
         )
 
         self.assertEqual(serialized, expected)
 
+    def test_json_decode(self):
+        serialized_json = json.dumps(
+            {
+                "class": "CategoricalColumn",
+                "data": {
+                    "name": "0",
+                    "col_index": np.nan,
+                    "sample_size": 0,
+                    "metadata": dict(),
+                    "times": defaultdict(),
+                    "thread_safe": True,
+                    "_categories": defaultdict(int),
+                    "_CategoricalColumn__calculations": dict(),
+                    "_top_k_categories": None,
+                },
+            }
+        )
+
+        deserialized = decode_column_profiler(serialized_json)
+        exepcted = CategoricalColumn("0")
+
+        test_utils.assert_profiles_equal(deserialized, exepcted)
+
+    def test_json_decode_after_update(self):
+        # Actual deserialization
+        serialized = json.dumps(
+            {
+                "class": "CategoricalColumn",
+                "data": {
+                    "name": None,
+                    "col_index": np.nan,
+                    "sample_size": 12,
+                    "metadata": {},
+                    "times": {"categories": 0.0},
+                    "thread_safe": True,
+                    "_categories": {"c": 5, "b": 4, "a": 3},
+                    "_CategoricalColumn__calculations": {},
+                    "_top_k_categories": None,
+                },
+            }
+        )
+        deserialized = decode_column_profiler(serialized)
+
+        # Build expected CategoricalColumn
+        df_categorical = pd.Series(
+            [
+                "a",
+                "a",
+                "a",
+                "b",
+                "b",
+                "b",
+                "b",
+                "c",
+                "c",
+                "c",
+                "c",
+                "c",
+            ]
+        )
+        profile = CategoricalColumn(df_categorical.name)
```

Comment:
Instead of naming it by the df, let's make something up that we know is unique to ensure it calc's it.

```python
profile = CategoricalColumn("Fake Profile Name")
```

however, your current tests the case of `None` which is a good edge case